contracts/Registry.vy

```diff
@@ -683,6 +848,15 @@ def _get_new_pool_decimals(_coins: address[MAX_COINS], _n_coins: uint256) -> uin
 def _remove_market(_pool: address, _coina: address, _coinb: address):
     key: uint256 = bitwise_xor(convert(_coina, uint256), convert(_coinb, uint256))
     length: uint256 = self.market_counts[key] - 1
+    if length == 0:
+        indexes: uint256 = self.coin_swap_indexes[key]
+        if convert(_coina, uint256) < convert(_coinb, uint256):
+            self._unregister_coin_pair(_coina, _coinb, indexes % 2 ** 128)
+            self._unregister_coin_pair(_coinb, _coina, shift(indexes, -128))
+        else:
```

Comment:
This feels pretty :vomiting_face: ... alternative was to bloat up the `_unregister_coin_pair` fn, to do deregistration in one call, but readability takes a hit.